### ISSUE TITLE
Users expected to select from a list of department and organizations

### ISSUE THREAD

**Issue by connorscarolyns** [cds-snc/platform-forms-client#1520]
Although government employees would be comfortable with a long dropdown list, it's recommended to look deeper into this issue to see if there is a more user-friendly pattern that can be applied; the design system might be able to help here.

 The design for this is pretty straightforward, but more needs to be understood on who should own and maintain this pattern.

### (Additional) acceptance criteria:
- [ ] Once this is merged in, we can flag to Support to follow up on [ticket 17129](https://cds-snc.freshdesk.com/a/tickets/17129)

**Comment by connorscarolyns** [comment]
The CRM team maintains a list of departments and has implemented a typeahead department selection component with Notify.

**Comment by connorscarolyns** [comment]
This would be a good one for UI / usability improvement @dsamojlenko @Abi-Nada

**Comment by connorscarolyns** [comment]
Beth (she/her)
  10:09 AM
https://github.com/cds-snc/gc-organisations
https://github.com/cds-snc/gc-organisations-qa
Pat (in core) and Stephen M (on leave?) both worked on this (edited) 
10:10
Also we had asked Abdul if he would build a reusable one, not sure if that ever started... might be worth checking with that team to see how you might save duplication?

**Comment by connorscarolyns** [comment]
Link to Figma design: https://www.figma.com/file/gkmd5aQpcnjuPeAT51oIFs/v2.-Add-modal?type=design&node-id=2349%3A3299&mode=design&t=GSPBpJn4mRFdy57H-1

**Comment by anikbrazeau** [comment]
[FIGMA](https://www.figma.com/file/gkmd5aQpcnjuPeAT51oIFs/v2.-Add-modal?type=design&node-id=2349-3299&mode=design&t=s8bPAVx7HINQmn3Q-4)

<img width="582" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMndaQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--7100fb9ca5b42be97b46afb2be5a29cd17835161/Screenshot%202024-01-31%20at%204.18.42%E2%80%AFPM.png" alt="Screenshot 2024-01-31 at 4.18.42 PM.png" />

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMjBaQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--0100d68a7d29c1eade82e64238d41548045c9ee1/Screenshot%202024-01-31%20at%204.19.02%E2%80%AFPM.png" alt="Screenshot 2024-01-31 at 4.19.02 PM.png" />

<img width="2000" src="https://api.zenhub.com/attachedFiles/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBMjhaQXc9PSIsImV4cCI6bnVsbCwicHVyIjoiYmxvYl9pZCJ9fQ==--b7e1dd1916fde3c940969524f455dd03564e71b7/Screenshot%202024-01-31%20at%204.19.16%E2%80%AFPM.png" alt="Screenshot 2024-01-31 at 4.19.16 PM.png" />

**Comment by anikbrazeau** [comment]
https://github.com/cds-snc/platform-forms-client/pull/3139

**Comment by Abi-Nada** [comment]
Work has been merged into Staging. We'll close this card and create new issues for any bugs.
